device/tlb.cpp

```diff
@@ -7,15 +7,16 @@
 namespace tt::umd {
 
 bool tlb_data::check(const tlb_offsets &offset) const {
-    return local_offset > ((1 << (offset.x_end - offset.local_offset)) - 1) |
-           x_end > ((1 << (offset.y_end - offset.x_end)) - 1) | y_end > ((1 << (offset.x_start - offset.y_end)) - 1) |
-           x_start > ((1 << (offset.y_start - offset.x_start)) - 1) |
-           y_start > ((1 << (offset.noc_sel - offset.y_start)) - 1) |
-           noc_sel > ((1 << (offset.mcast - offset.noc_sel)) - 1) |
-           mcast > ((1 << (offset.ordering - offset.mcast)) - 1) |
-           ordering > ((1 << (offset.linked - offset.ordering)) - 1) |
-           linked > ((1 << (offset.static_vc - offset.linked)) - 1) |
-           static_vc > ((1 << (offset.static_vc_end - offset.static_vc)) - 1);
+    return local_offset > ((1ULL << (offset.x_end - offset.local_offset)) - 1) |
```

Comment:
Is it possible to easily cover this by any existing/new test?

Reply:
Yes although I'm reluctant to spend time writing such tests because I don't think tlb.cpp and the architecture_implementation business is a good way to do this, plus all of this belongs in the kernel driver anyway.